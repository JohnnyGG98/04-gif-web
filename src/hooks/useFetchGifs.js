import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (categoria) => {
  const [state, setstate] = useState({
    data: [],
    loading: true
  });

  useEffect(() => {
    getGifs(categoria)
      .then( data => {
        setTimeout(() => {
          setstate({
            data: data,
            loading: false
          });
        }, 3000);
      });
  }, [categoria]);

  return state;
}