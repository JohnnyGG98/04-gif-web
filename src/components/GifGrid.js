import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({ categoria }) => {
  // const apiKey = 'RuLBOmXSEY5rCfTT2sEEWcHSaIC737EN';

  const {data: images, loading} = useFetchGifs(categoria);
  
  return (
    <>
      <h3 className="animate__animated animate__bounce">{categoria}</h3>

      { loading && <p className="animate__animated animate__flash">Cargando...</p> }

      <div className="card-grid">
        {images.map(img => (          
          <GifGridItem 
            key={img.id} 
            {...img}
            />
        ))}

      </div>
    </>
  );
}

GifGrid.propTypes = {
  categoria: PropTypes.string.isRequired
}

export default GifGrid;