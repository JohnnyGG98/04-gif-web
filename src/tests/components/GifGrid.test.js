import '@testing-library/jest-dom';

import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas | <GifGrid />', () => {

  const cat = 'Hunter';

  test('Render ', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow( <GifGrid categoria={cat} /> ) 
    expect(wrapper).toMatchSnapshot();
  });

  test('Return informarcion', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://',
        title: 'TITLE'
      }
    ];
    
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow( <GifGrid categoria={cat} /> ) 

    expect(wrapper).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe( false );
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
  });

});