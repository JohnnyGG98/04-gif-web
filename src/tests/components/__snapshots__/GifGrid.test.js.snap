// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas | <GifGrid /> Render  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Hunter
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas | <GifGrid /> Return informarcion 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Hunter
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="TITLE"
      url="https://"
    />
  </div>
</Fragment>
`;
